jax_unirep/evotuning.py

```diff
@@ -160,10 +161,122 @@ def length_batch_input_outputs(
     return xs, ys
 
 
+def fit_length_batches(
+    params: Dict, state, update, get_params, xs, ys, step_size: float = 0.001,
+) -> Dict:
+    """
+    Perform one epoch of weight updates using length batching.
```

Comment:
I realized here that a brief sentence describing "length batching" is probably going to be helpful for future selves, because the term "length batching" isn't widely-used enough for one to remember it easily.

Here's a straw man starter:

```
In length batching, we first "batch" sequences together according to length.
Sequences of the same length go into the same batch.
Gradient updates happen batch-by-batch.
```